### PR TITLE
Add fparser and stylist recipes

### PR DESCRIPTION
**Pull request by matthewrmshin**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] ~If static libraries are linked in, the license of the static library is packaged.~
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by MatthewHambley** [comment]
This post is to confirm that I am willing to be listed as a maintainer.

**Comment by matthewrmshin** [comment]
(Awaiting agreement to add [fparser](https://pypi.org/project/fparser/) recipe in this PR.)

**Comment by rupertford** [comment]
I confirm that I am happy to be listed as a maintainer.

**Comment by matthewrmshin** [comment]
@conda-forge-admin, please ping team

This PR is ready for review by @conda-forge/help-python

File: recipes/fparser/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: 2b5051e43a3e782c30f85cacabdb597737b8ec1205a91032d7584b0e61f27a2f
+
+build:
+  number: 0
```
Comment (npavlovikj) [inline recipes/fparser/meta.yaml]: I believe this recipe can be `noarch`, same as the other one?

File: recipes/fparser/LICENSE [hunk 2]
```diff
@@ -0,0 +1,63 @@
+Modified work Copyright (c) 2017 Science and Technology Facilities Council
```
Comment (npavlovikj) [inline recipes/fparser/LICENSE]: Looks like the license file is available in the pypi tarball, so please delete the file from here:
```
fparser-0.0.10/LICENSE
x fparser-0.0.10/doc/
x fparser-0.0.10/doc/conf.py
x fparser-0.0.10/doc/Makefile
x fparser-0.0.10/doc/introduction.rst
```

File: recipes/stylist/LICENSE [hunk 3]
```diff
@@ -0,0 +1,29 @@
+BSD 3-Clause License
```
Comment (npavlovikj) [inline recipes/stylist/LICENSE]: Please delete this file as well, the LICENSE file itself is included in the pypi tarball.

**Comment by matthewrmshin** [comment]
Hi @npavlovikj Thank you for the review. 😃 I have now fixed the issues you have raised. (I must have misread the document about the requirement of the license file in the recipe.)

File: recipes/stylist/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+build:
+  entry_points:
+    - stylist=stylist.__main__:main
+  noarch: python
+  number: 0
+  script: "{{ PYTHON }} -m pip install . --no-deps --ignore-installed -vv"
+
+requirements:
+  host:
+    - fparser
```
Comment (chrisburr) [inline recipes/stylist/meta.yaml]: Do you need `fparser` in `host`? Normally `pip` and `python` are enough.
```suggestion
```

File: recipes/fparser/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    - fparser2=fparser.scripts.fparser2:main
+  noarch: python
+  number: 0
+  script: "{{ PYTHON }} -m pip install . --no-deps --ignore-installed -vv "
+
+requirements:
+  host:
+    - pip
+    - python
+    - six
```
Comment (chrisburr) [inline recipes/fparser/meta.yaml]: Do you need `six` in `host`? Normally `pip` and `python` are enough.
```suggestion
```

File: recipes/fparser/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+    - fparser.one
+    - fparser.one.tests
+    - fparser.scripts
+    - fparser.two
+  commands:
+    - fparser2 --help
+
+about:
+  home: https://github.com/stfc/fparser
+  license: BSD
```
Comment (chrisburr) [inline recipes/fparser/meta.yaml]: SPDX identifiers are preferred:
```suggestion
  license: BSD-3-Clause
```
(I guess it's three clause but please check!)

File: recipes/stylist/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+
+test:
+  imports:
+    - stylist
+  commands:
+    - stylist --help
+
+about:
+  home: https://github.com/MetOffice/stylist
+  license: BSD
```
Comment (chrisburr) [inline recipes/stylist/meta.yaml]: Same:
```suggestion
  license: BSD-3-Clause
```

**Comment by chrisburr** [comment]
> I must have misread the document about the requirement of the license file in the recipe

Do you have a suggestion for better phrasing? It's a common misunderstanding :)

**Comment by matthewrmshin** [comment]
@chrisburr Thank you for the review. Your review comments should be addressed in the latest changeset.

> Do you have a suggestion for better phrasing? It's a common misunderstanding :)

On the license text, I guess we can probably have a statement to say that "a license file in the package is *sufficient* to satisfy the recipe requirement"?

**Comment by matthewrmshin** [comment]
🎉

**Pull request merged**